LinkedListImplementation/LinkedListImplementation.cs

```diff
@@ -76,6 +76,134 @@ public T GetMax()
             return max;
         }
 
+        public void Add(T value)
+        {
+            if (Tail == null)
+            {
+                Head = new Node<T>(value);
+                Tail = Head;
+            }
+            else
+            {
+                var current = new Node<T>(value);
```

Comment:
Add a separate method that stores all the same logic from Add and AddStart methods and accepts delegate where we can pass different logic